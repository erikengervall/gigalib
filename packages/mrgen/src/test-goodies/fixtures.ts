import { MRGenOptions } from '../types/mrgen-options';

export const createMockMRGenOptions = (
  overrides?: Partial<MRGenOptions>,
): { mrGenOptions: MRGenOptions } => {
  return {
    mrGenOptions: {
      data: {
        queryParams: true,
        payload: true,
        params: true,
        responseBody: true,
      },

      httpMethod: 'POST',

      shared: {
        apiDir: 'packages/shared/src/api',
        file: 'packages/shared/src/api/v1/user/get.ts',
        buildCommand: 'yarn workspace shared ci:build',
      },
      backend: {
        file: 'packages/backend/src/handlers/v1/user/get-handler.ts',
      },
      frontend: {
        file: 'packages/frontend1/src/api/hooks/v1-user-get.ts',
      },
      ...overrides,
    },
  };
};
