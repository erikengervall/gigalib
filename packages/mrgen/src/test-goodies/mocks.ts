import cp from 'node:child_process';
import fs from 'node:fs';

export const mockExecSync = cp.execSync as jest.Mock;
export const mockMkdirSync = fs.mkdirSync as jest.Mock;
export const mockWriteFileSync = fs.writeFileSync as jest.Mock;
