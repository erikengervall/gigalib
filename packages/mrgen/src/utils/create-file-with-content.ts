import fs from 'node:fs';
import path from 'node:path';

export function createFileWithContent(filePath: string, content: string) {
  const filePathDirName = path.dirname(filePath);
  console.debug('mkdirSync', filePathDirName);
  fs.mkdirSync(filePathDirName, { recursive: true });

  console.debug('writeFileSync', filePath);
  fs.writeFileSync(filePath, content, 'utf-8');
}
