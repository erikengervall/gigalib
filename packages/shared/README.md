# @engervall/shared

Utilities and stuff.
