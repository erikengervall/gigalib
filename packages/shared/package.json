{
  "name": "@engervall/shared",
  "version": "1.0.6-alpha.0",
  "main": "dist/index.js",
  "repository": "git@github.com:erikengervall/gigalib.git",
  "author": "Erik Engervall <erik.engervall@gmail.com>",
  "scripts": {
    "----- CI": "",
    "ci:build": "tsc",
    "ci:format": "prettier --check './src/**/*.{ts,tsx}' --ignore-path ../../.prettierignore",
    "ci:lint": "eslint 'src/**/*.{ts,tsx}'",
    "ci:test:unit": "jest",
    "----- LOCAL DEV": "",
    "dev": "rm -rf dist && tsc --watch",
    "----- MISC": "",
    "format:write": "prettier --write './src/**/*.{ts,tsx}'",
    "clean": "rm -rf dist",
    "reinstall": "yarn clean && rm -rf node_modules"
  },
  "dependencies": {
    "qs": "^6.11.2",
    "zod": "^3.21.4",
    "zod-validation-error": "^1.3.1"
  }
}
