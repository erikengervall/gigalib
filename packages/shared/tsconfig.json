{
  "extends": "../../tsconfig.json",
  "compilerOptions": {
    "declaration": true,
    "lib": ["es2021"],
    "module": "CommonJS",
    "noEmit": false,
    "sourceMap": true,
    "target": "ES6",

    "baseUrl": "src",
    "outDir": "dist",
    "rootDir": "src"
  },
  "include": ["src", "../../scripts/change-case.ts"],
  "exclude": [
    "_warmup",
    ".cache",
    ".next",
    ".serverless",
    ".vscode",
    ".vscode",
    ".webpack",
    "*.dist*",
    "*.spec*",
    "*.test*",
    "build",
    "coverage",
    "cypress-coverage",
    "dist",
    "jest-coverage",
    "node_modules"
  ]
}
