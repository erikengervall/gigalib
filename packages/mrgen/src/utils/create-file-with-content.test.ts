import { logSnapshots } from '@engervall/shared';

import { mockMkdirSync, mockWriteFileSync } from '../test-goodies/mocks';
import { createFileWithContent } from './create-file-with-content';

it('should create a file with the given content', () => {
  // Arrange
  const filePath = 'test-file-path.txt';
  const content = `
  test content
  1
  2
  3
  `;

  // Act
  createFileWithContent(filePath, content);

  // Assert
  expect(mockMkdirSync.mock.calls).toMatchSnapshot('fs.mkdirSync');
  expect(mockWriteFileSync.mock.calls).toMatchSnapshot('fs.writeFileSync');
  logSnapshots();
});
