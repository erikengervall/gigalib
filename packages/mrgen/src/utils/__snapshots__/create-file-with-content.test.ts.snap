// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should create a file with the given content: aggregateLogProxySpy 1`] = `
[
  {
    "debug": [
      "mkdirSync",
      ".",
    ],
  },
  {
    "debug": [
      "writeFileSync",
      "test-file-path.txt",
    ],
  },
]
`;

exports[`should create a file with the given content: fs.mkdirSync 1`] = `
[
  [
    ".",
    {
      "recursive": true,
    },
  ],
]
`;

exports[`should create a file with the given content: fs.writeFileSync 1`] = `
[
  [
    "test-file-path.txt",
    "
  test content
  1
  2
  3
  ",
    "utf-8",
  ],
]
`;
